educator/src/Dscp/Educator/Web/Student/Queries.hs

```diff
@@ -172,18 +170,18 @@ getLastAssignmentSubmission student assignH = do
         limit     1
     |]
 
-getAssignment
+studentGetAssignment
     :: (MonadStudentAPIQuery m, WithinSQLTransaction)
-    => Student -> Hash Core.Assignment -> m Assignment
-getAssignment student assignH = do
+    => Student -> Hash Assignment -> m AssignmentStudentInfo
+studentGetAssignment student assignH = do
     massign <-
         query queryText (assignH, student)
         >>= listToMaybeWarn "assignments"
-    (aCourseId, aContentsHash, assignType, aDesc) <-
+    (aiCourseId, aiContentsHash, assignType, aiDesc) <-
         pure massign `assertJust` AssignmentDoesNotExist assignH
-    let IsFinal aIsFinal = assignType ^. assignmentTypeRaw
-    aLastSubmission <- getLastAssignmentSubmission student assignH
-    return Assignment{ aHash = assignH, .. }
+    let IsFinal aiIsFinal = assignType ^. assignmentTypeRaw
+    aiLastSubmission <- studentGetLastAssignmentSubmission student assignH
+    return AssignmentStudentInfo{ aiHash = assignH, .. }
```

Comment:
Please, don't use `RecordWildCards` this way.
The data flow becomes very obscure with it. (You can rename `assignH` to `aiHash` to improve obscurity)

Also, there is `Id` type family from `Dscp.Util.HasId` typeclass; please use it when referring to entities, don't use their hashes. The `Id a` is the `Hash a` for most our types.

if there is no instance for some new entity, please make one.

It even has sane instances for all good old types once lived in `Dscp.Core.Types`.

Reply:
The reason the dataflow is obscure is that you meet `FooBar { .. }` and the flow of reading breaks: you have to go up in the text to collect everything that looks like `fbQux`.

You can't reconstruct data going in in this line just by looking on it.

Reply:
But the point of this function and all others in this module is entirely to construct a response, seems to be justified use of `RecordWildCards`

Reply:
I'm personally against usage of this extension. But don't insist here.